startup/gui/graphEditor.py

```diff
@@ -233,9 +234,79 @@ def __drop( graphGadget, event ) :
 
 	return True
 
+##########################################################################
+# Scene location drop handler
+##########################################################################
+
+def __dropLocationData( event ) :
+
+	if (
+		not isinstance( event.data, IECore.StringVectorData ) or
+		len( event.data ) != 1 or
+		not event.data[0].startswith( "/" )
+	) :
+		return None
+
+	scene = None
+	sourceEditor = event.sourceWidget.ancestor( GafferUI.Editor )
+	if isinstance( sourceEditor, GafferUI.Viewer ) :
+		if isinstance( event.sourceGadget, GafferSceneUI.SceneGadget ) :
+			scene = sourceEditor.view()["in"].getInput()
+	elif isinstance( sourceEditor, GafferSceneUI.HierarchyView ) :
+		scene = sourceEditor.scene()
+
+	if scene is None :
+		return None
+
+	return {
+		"path" : event.data[0],
+		"scene" : scene,
+		"context" : sourceEditor.getContext(),
+	}
+
+def __locationDragEnter( graphGadget, event ) :
+
+	if __dropLocationData( event ) is None :
+		return False
+
+	GafferUI.Pointer.setCurrent( "target" )
+	return True
+
+def __locationDragLeave( graphGadget, event ) :
+
+	if __dropLocationData( event ) is not None :
+		GafferUI.Pointer.setCurrent( None )
```

Comment:
Super minor nitpick, but we're not restoring the previous Pointer here, so if you were to drag a single object across the Graph Editor to a floating Node Editor, the cursor would begin with "objects", change to "target" as you drag across the Graph Editor, and then switch to `None` as you leave the Graph Editor and enter the Node Editor, which then feels like you've lost the drag...

Reply:
Thanks for catching that - should be fixed by https://github.com/GafferHQ/gaffer/pull/5772/commits/eb8713c814d4c25998c1680eb489f45efb1dd735. I've rebased everything to fix the merge conflict, but the only new changes are the ones in that commit.

Ack. No. I knew there was a wider problem with this sort of thing. Try this :

- Drag from HierarchyView into GraphEditor, getting the target icon.
- Continue drag over onto a PathFilter, getting the object assignment icon.
- Drag off PathFilter, getting back the target icon.
- Now drag out of GraphEditor into NodeEditor. Now we're back to the object assignment icon rather than the original icon.

I don't think this is really resolvable at the level of an individual drag handler. There's also the problem that `dragLeave` is emitted on the old widget _after_ `dragEnter` on the new one, so you can end up with pointer-fighting that way too.

Reply:
Ugh. I did worry that this might not be straightforward when I noticed it. Cheap and cheerful - could we just set the pointer back to "objects" rather than `None` and leave a todo about the wider issue?

Reply:
> There's also the problem that `dragLeave` is emitted on the old widget _after_ `dragEnter` on the new one, so you can end up with pointer-fighting that way too.

I was confused as to why this hadn't become a problem for use when dragging onto a PathFilter - I thought we would be clobbering the pointer assigned by PathFilter. Turns out we're OK because PathFilter uses `dragMove` to set the pointer, so it can change it when `Ctrl` or `Shift` is pressed. But does demonstrate that trying to restore the pointer in `dragLeave` is flawed in the general case.

> Cheap and cheerful - could we just set the pointer back to "objects" rather than None and leave a todo about the wider issue?

I've done that and squashed it into https://github.com/GafferHQ/gaffer/pull/5772/commits/42f05b5245ccb4d61878541748598a7447cd6539.